Change index.html -> index.md and make intro text clearer

jnewbery opened the pull request:
Closes #69 

The current intro text is a block of prose. This splits it out and makes the purpose of the review club more obvious.

jnewbery commented on index.md:
```diff
@@ -0,0 +1,113 @@
+<span style="font-size:1.25em; font-weight:bold">What's it for?</span> To
+help newer contributors learn about the Bitcoin Core review process. Review Club
+is *not* primarily intended to help open PRs get merged (although that might be
+a nice side-effect). The Review Club will give participants the tools and
```
This seems to be a frequent misunderstanding. People often think that Review Club is to focus review on a PR to help it gets merged, much like the "high-priority for review" list. That's not a goal of review club, which I want to make explicit here.

jonatack replied:
Oh, interesting. I could see a potential incidental effect of the additional activity inciting experienced reviewers to look at a PR sooner than they might otherwise have, but it was clear to me that it's not a primary goal of the club, and I reckoned review weight for maintainers is more about quality than quantity anyway; the [peer review
process](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review) makes that clear.

jnewbery commented on index.md:
```diff
@@ -0,0 +1,113 @@
+knowledge they need in order to take part in the Bitcoin Core review process on
+Github.
+
+<span style="font-size:1.25em; font-weight:bold">Who should take part?</span>
```
Moved to a css class.

jnewbery commented on index.md:
```diff
@@ -0,0 +1,113 @@
+</span>
+</div>
+{%- endif -%}
+{% endfor %}
```
When I left the indentation in, this was formatted as pre-rendered code.

jnewbery replied:
ok, I've managed to restore the indentation

jonatack replied:
nice!

jnewbery commented:
> I miss the "A weekly review club for Bitcoin Core PRs" subtitle. It could be a bit smaller but I think it's a pretty good home page subtitle.

restored

> If we keep this format, I'd suggest these general points in this order: who, what, where, why, how (not necessarily verbatim as titles, but the idea).

Any particular reason beyond aesthetics? I think the current order makes sense. Putting the 'who' before the 'what' and 'why' doesn't make sense to me.

> "What's it for?" and "Why do this?" seem overlapping and redundant; they both discuss Why?

I've renamed "why do this?" to "What's the benefit for participants?". Does that make it clearer? There is a bit of redundancy, but the first is an overview of why the review club exists, and the second is an argument for why an individual should take part. Any suggested tweaks are welcome!

> The repeated style tags could be moved out to the css file.

Done!

> I'm not sure converting an infrequently-changed page like this one to markdown is worth it; losing the indentation is a loss of readability and could be a source of future errors.

I prefer markdown in almost all cases, and I've reinstated the indentation for the liquid loops now, which hopefully addresses your concerns about readability.

> It's not easy to tell from looking at the html-to-markdown conversion, but it does appear that everything from Upcoming Meetings on down is unchanged in rendering, as expected IIUC.

You can test by:

- checkout master and run `make preview`
- move the `_site` folder to `_site.bak`
- checkout this branch and run `make preview`
- run your difftool over `_site.bak` and `_site` directories

jonatack commented on index.md:
```diff
@@ -0,0 +1,110 @@
+### A weekly review club for Bitcoin Core PRs
+
+<span class="question">What is this?</span> A weekly
+club for reviewing Bitcoin Core PRs in the Freenode `#bitcoin-core-pr-reviews`
```
Perhaps "IRC" here, and the more detailed "Freenode" in the How section.

jnewbery replied:
done

jonatack commented on index.md:
```diff
@@ -0,0 +1,110 @@
+- Clone the repo, checkout and build the PR branch and run all tests.
+- Review the code changes and read the comments on the PR.
+- Make a note of any questions you want to ask.
+- Join the IRC channel at **18:00 UTC on wednesday**.
```
s/wednesday/Wednesday/?

jnewbery replied:
done

jonatack commented on index.md:
```diff
@@ -0,0 +1,110 @@
+- Make a note of any questions you want to ask.
+- Join the IRC channel at **18:00 UTC on wednesday**.
+
+<span class="question">Who runs this?</span> jnewbery
```
s/jnewbery/John Newbery/? Using your IRC/GH nick here might seem odd.

jnewbery replied:
I prefer to use my nick here than full name. I've added a link to my github profile in case it's not obvious who this is referring to.

jonatack commented on index.md:
```diff
@@ -0,0 +1,110 @@
+    <span class="host">hosted by
+      <a class="host" href="/meetings-hosts/#{{post.host}}">{{ post.host }}</a>
+    </span>
+  </div>{%- endif -%}
```
nit: I think it would be easier to parse visually if the Liquid conditional and HTML `div` markup here on lines 52 and 60 (and similar cases below) were on separate lines and indentations -- in general, separate the Liquid from the HTML.

jnewbery replied:
If I do that then jekyll renders this as code. If you can find a way to have these on separate lines _and_ indented and have jekyll render it correctly, then I'm happy to take it.

jonatack commented on assets/css/all.sass:
```diff
@@ -53,3 +53,7 @@ a
   background: #ddd
+
+.question
+  font-size: 1.25em
```
nit: perhaps a bit smaller (maybe same size as the ### subtitle is in this PR)

jnewbery replied:
changed to 1.2

jonatack commented:
> > If we keep this format, I'd suggest these general points in this order: who, what, where, why, how (not necessarily verbatim as titles, but the idea).
> 
> Any particular reason beyond aesthetics? I think the current order makes sense. Putting the 'who' before the 'what' and 'why' doesn't make sense to me.
> 

That's true.

I had it impressed into me in elementary school that a news story should cover these items and that leading with "who", the human element, is what generally interests people the most. So I return to that thought every time :smile:

jnewbery commented:
Thanks for the review @jonatack ! I've addressed your comments. I've also tightened up the text a bit and hopefully made the distinction between the *What’s it for?* and *What’s the benefit for participants?* sections a bit more obvious.

I prefer this to the current text because it optimizes for first-time visitors to the page by explaining what the review club is as clearly and concisely as I could. There's possibly a bit more text and screen real estate used than the current version, but I think it's clearer.

jnewbery commented:
I'm going to squash and merge this now. Any further changes/improvements are still very welcome in follow-up PRs.
